nightshift/comms/sierra_search_platform.py

```diff
@@ -0,0 +1,337 @@
+                return "deleted"
+            # if bib orignated from Worldcat asssume full bib
+            if "ss_marc_tag_003" in data and data["ss_marc_tag_003"] == "OCoLC":
+                return "full-bib"
+
+            # print material with call number tag - assume full bib
+            # exclude electronic resources
+            if "call_number" in data and not is_eresource_callno(data["call_number"]):
+                return "full-bib"
+
+        # assume at this point it must be a brief bib
+        return "brief-bib"
+
+    def _determine_nyp_bib_status(self) -> str:
+        """
+        Determines status of record in NYPL Sierra.
+
+        Returns:
+            'bief-bib', 'full-bib' or 'deleted' status
+        """
+        data = self.json_response["data"]
+        if data["deleted"]:
+            return "deleted"
+        else:
+            # check first if Sierra bib came from the Worldcat;
+            # and catch here upgraded/enhanced electronic resources
+            for field in data["varFields"]:
+                if field["marcTag"] == "003" and field["content"] == "OCoLC":
+                    return "full-bib"
+
+            # print material full bibs may come from other sources than
+            # Worldcat (no or diff content of the '003' tag);
+            # brief bibs lack call numbers
+            for field in data["varFields"]:
+                if field["marcTag"] == "091":  # filter out electronic resources
+                    if not is_eresource_callno(field["subfields"][0]["content"]):
+                        return "full-bib"
+
+            # assume response failing previous clauses is a brief bib
+            return "brief-bib"
+
+    def _bpl_suppression(self) -> bool:
+        """
+        Checks if BPL Sierra bib is suppressed from display.
+
+        Returns:
+            bool
+        """
+        if len(self.json_response["response"]["docs"]) == 0:
+            return False
+        elif self.json_response["response"]["docs"][0]["suppressed"]:
+            return True
+        else:
+            return False
+
+    def _nyp_suppression(self) -> bool:
+        """
+        Checks if NYPL Sierra bib is suppressed from display.
+
+        Returns:
+            bool
+        """
+        if self.response.status_code == 404:
+            return False
+        elif self.json_response["data"]["suppressed"]:
+            return True
+        else:
+            return False
+
+
+class NypPlatform(PlatformSession):
+    def __init__(self) -> None:
+        """
+        Authenticates and opens a session with NYPL Platform.
+        Relies on credentials stored in evironmental variables.
+        """
+        client_id, client_secret, oauth_server, target = self._get_credentials()
+        token = self._get_token(client_id, client_secret, oauth_server)
+        agent = f"{__title__}/{__version__}"
+
+        PlatformSession.__init__(self, authorization=token, agent=agent, target=target)
```

Comment:
I think the more "pythonic" implementation would be to use `super()` here!

Reply:
Thanks!